Sources/PubNub/EventEngine/Presence/PresenceTransition.swift

```diff
@@ -26,15 +26,15 @@ class PresenceTransition: TransitionProtocol {
     case .joined:
       return configuration.heartbeatInterval > 0
     case .left:
-      return !(state is Presence.HeartbeatInactive)
+      return true
     case .heartbeatSuccess:
       return state is Presence.Heartbeating
     case .heartbeatFailed:
       return state is Presence.Heartbeating
     case .timesUp:
       return state is Presence.HeartbeatCooldown
     case .leftAll:
-      return !(state is Presence.HeartbeatInactive)
+      return true
```

Comment:
So, we want to transit even if heartbeat loop is not active?

Reply:
Yes. You're right, the heartbeat loop is inactive if the user subscribes to some channel without Presence. However, we still have to trigger `.leave` invocation as a part of the unsubscribe cleanup